### PR TITLE
Fix: simplify check for supported hooks

### LINKED ISSUE #47: Adjust rule for Valtio v2
In v2, the incompatibility with useMemo and so on will be resolved, https://github.com/pmndrs/valtio/pull/866
so, this plugin should avoid warning such usages with v2.

I wonder if there's a way in eslint to change the behavior based on Valtio versions. Otherwise, we can simply upgrade, which is reasonable for simplicity too.

@barelyhuman Can you work on it? It's not in a hurry.

### PR DESCRIPTION
**Pull request by barelyhuman**
Closes #47

### CODE REVIEW

File: tests/StateSnapshot.test.js [hunk 1]
```diff
@@ -298,6 +298,20 @@ ruleTester.run('state-snapshot-rule', rule, {
...
+      return s.a1.b.c
+    },[s.a1.b.c])
+  }`,
+    `
+  const foo = proxy({ foo: 123 });
+  function useExample2(s) {
+    const val = useMemo(()=>{
+      return foo.foo
+    },[foo.foo])
+  }`,
```
Comment (dai-shi) [inline tests/StateSnapshot.test.js:314]: In addition to that, this is valid too.
```js
  const foo = proxy({ foo: 123 });
  function useExample2(s) {
    const val = useMemo(()=>{
      return foo.foo
    },[foo])
  }
```

File: tests/StateSnapshot.test.js [hunk 2]
```diff
@@ -298,6 +298,20 @@ ruleTester.run('state-snapshot-rule', rule, {
       </>
     );
   })`,
+    `
+  function useExample2(s) {
+    const {b: {c} } = useSnapshot(s.a1);
+    const val = useMemo(()=>{
+      return s.a1.b.c
+    },[s.a1.b.c])
+  }`,
```
Comment (dai-shi) [inline tests/StateSnapshot.test.js:307]: That's actually not good.
The deps should come from snapshots.
```js
  function useExample2(s) {
    const {b: {c} } = useSnapshot(s.a1);
    const val = useMemo(()=>{
      return c
    },[c])
  }
```

File: tests/StateSnapshot.test.js (hunk 1 above)
Comment (barelyhuman) [reply tests/StateSnapshot.test.js:321]: Updated

File: tests/StateSnapshot.test.js (hunk 2 above)
Comment (barelyhuman) [reply tests/StateSnapshot.test.js:307]: Updated

**Review by dai-shi — APPROVED** [review]
LGTM

**Pull request merged**